### PR TITLE
Move fulfillment warehouse selection to fulfilll page

### PR DESCRIPTION
**Pull request by orzechdev**
It:
- moves fulfillment warehouse selection from the order page to the fulfill order page,
- divides warehouse selection on the fulfill order page per each product line, with preselected warehouse according to given line allocation,
- removes tracking number input from the fulfill order page.

**PR intended to be tested with API branch:**  main

### Screenshots

Removed warehouse selection in order page:

Screenshot 2022-07-28 at 22 27 25

Added warehouse selection in fulfill order page:

Screenshot 2022-07-29 at 12 16 09

Adjusted warehouse selection modal:

Screenshot 2022-07-28 at 22 27 18

Removed tracking number input:

Screenshot 2022-07-28 at 22 26 41

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://marketplace-gray.vercel.app/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-7723 Orders - selected warehouse with the highest available quantity.](https://app.clickup.com/t/2549495/SALEOR-7723)

File: src/orders/components/OrderFulfillLine/OrderFulfillLine.tsx [hunk 1]
```diff
@@ -141,6 +158,20 @@ export const OrderFulfillLine: React.FC<OrderFulfillLineProps> = props => {
...
       </TableCell>
+      <TableCell className={classes.colWarehouse}>
+        <IconButton
+          onClick={onWarehouseChange}
+          className={classes.warehouseButton}
+          data-test-id="select-warehouse-button"
+        >
+          <div className={classes.warehouseButtonContent}>
+            <div className={classes.warehouseButtonContentText}>
+              {lineFormWarehouse.name ?? <Skeleton />}
```
Comment (Droniu) [inline src/orders/components/OrderFulfillLine/OrderFulfillLine.tsx:169]:
```suggestion
              {lineFormWarehouse?.name ?? <Skeleton />}
```

File: src/orders/components/OrderFulfillPage/OrderFulfillPage.tsx [hunk 2]
```diff
@@ -87,22 +96,37 @@ const OrderFulfillPage: React.FC<OrderFulfillPageProps> = props => {
...
+            if (
+              !prevAllocation ||
+              prevAllocation.quantity < allocation.quantity
+            ) {
+              return allocation;
+            }
+            return prevAllocation;
+          },
+          null,
+        );
```
Comment (Droniu) [inline src/orders/components/OrderFulfillPage/OrderFulfillPage.tsx:114]: I'd specify type of `highestQuantityAllocation` explicitly and move this to utils file

File: src/orders/components/OrderChangeWarehouseDialog/messages.ts [hunk 3]
```diff
@@ -7,8 +7,8 @@ export const changeWarehouseDialogMessages = defineMessages({
     description: "change warehouse dialog title",
   },
   dialogDescription: {
-    defaultMessage: "Choose warehouse you want to fulfill this order from",
-    id: "QWCh6/",
+    defaultMessage: "Choose warehouse from you want to fulfill {productName}",
```
Comment (Droniu) [inline src/orders/components/OrderChangeWarehouseDialog/messages.ts:10]:
```suggestion
    defaultMessage: "Choose warehouse from which you want to fulfill {productName}",
```

File: src/orders/components/OrderFulfillStockExceededDialogLine/OrderFulfillStockExceededDialogLine.tsx [hunk 4]
```diff
@@ -14,13 +14,17 @@ import { useStyles } from "../OrderFulfillStockExceededDialog/styles";
...
 
 const OrderFulfillStockExceededDialogLine: React.FC<OrderFulfillStockExceededDialogLineProps> = props => {
   const { line: genericLine, warehouseId, formsetData } = props;
 
-  const line = "orderLine" in genericLine ? genericLine.orderLine : genericLine;
+  const line = genericLine
+    ? "orderLine" in genericLine
+      ? genericLine.orderLine
+      : genericLine
+    : null;
```
Comment (Droniu) [inline src/orders/components/OrderFulfillStockExceededDialogLine/OrderFulfillStockExceededDialogLine.tsx:27]: I think you can just move 

```
  if (!genericLine) {
    return null;
  }
```
from lines 34-36 up here and then we can avoid nested ternary

File: src/orders/urls.ts [hunk 5]
```diff
@@ -126,7 +126,14 @@ export type OrderUrlDialog =
 
 export type OrderUrlQueryParams = Dialog<OrderUrlDialog> & SingleAction;
 
-export type OrderFulfillUrlQueryParams = Partial<{ warehouse: string }>;
+export enum OrderFulfillUrlFiltersEnum {
+  warehouseId = "warehouseId",
+  lineId = "lineId",
+}
```
Comment (Droniu) [inline src/orders/urls.ts:132]: we should avoid new enums, can we use union types here instead?

File: src/orders/utils/data.ts [hunk 6]
```diff
@@ -356,17 +360,49 @@ export const isLineAvailableInWarehouse = (
   return false;
 };
 
-export const transformFuflillmentLinesToStockInputFormsetData = (
+export const getLineAvailableQuantityInWarehouse = (
+  line: OrderFulfillLineFragment,
+  warehouse: WarehouseFragment,
+) => {
```
Comment (Droniu) [inline src/orders/utils/data.ts:366]:
```suggestion
): number => {
```

File: src/orders/views/OrderDetails/OrderNormalDetails/index.tsx [hunk 7]
```diff
@@ -48,7 +46,6 @@ import {
   OrderUrlQueryParams,
 } from "../../../urls";
 import { isAnyAddressEditModalOpen } from "../OrderDraftDetails";
-import { useDefaultWarehouse } from "./useDefaultWarehouse";
```
Comment (Droniu) [inline src/orders/views/OrderDetails/OrderNormalDetails/index.tsx:51]: This hook is probably used nowhere else, so this file can be deleted also

**Review by dominik-zeglen — APPROVED** [review]

File: src/orders/utils/data.ts (hunk 6 above)
Comment (orzechdev) [reply src/orders/utils/data.ts:366]: There is no need to add typing as the type is inferred automatically.

**Review by Droniu — APPROVED** [review]

**Pull request merged**